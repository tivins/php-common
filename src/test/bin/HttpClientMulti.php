<?php

use Tivins\Core\Net\Client;
use Tivins\Core\Net\ClientException;
use Tivins\Core\Net\ClientMulti;

require 'vendor/autoload.php';

$clients = new ClientMulti();
// Add clients: URL or Client
$clients->addClients(
    'https://example.com/',
    (new Client('https://httpbin.org/anything'))->postJSON(['yo' => 'lo']),
);

$duration = $clients->execute();
foreach ($clients->getClients() as $k => $client) {
    printf(
        "- #%d : %s, Code: %d, Size: %d bytes\n",
        $k, $client, $client->getCode(), strlen($client->getContent())
    );
}


