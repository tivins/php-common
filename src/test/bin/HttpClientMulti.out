- #45 : Tivins\Core\Net\Client#43, Code: 200, Size: 1256 bytes
- #46 : Tivins\Core\Net\Client#5, Code: 200, Size: 558 bytes
